### PR TITLE
Return result as to corresponding python value

### PR DESCRIPTION
**Pull request by SeaHOH**
Hello:
It's a nice project! I have been using it for a while.
For now, all the result is in `str` type, sometimes this is not  inconvenient to us, and I hope that you can consider this PR.

There are 3 changes:
1. Return result as to corresponding python value instead of string only.
    Method used: value => json => value
2. Use some Windows-only API.
3. Add exception type info to result, not only the message string.

### CODE REVIEW

File: PyChakra/__init__.py [hunk 1]
```diff
@@ -60,31 +60,73 @@ def __init__(self):
...
-    def eval(self, js_string):
-        js_source = ctypes.c_void_p()
-        self.chakraCore.JsCreateString("", len(""), point(js_source))
+    if sys.platform == "win32":
+        def eval(self, js_string, raw=False):
+            js_source = ctypes.c_wchar_p("")
+            js_script = ctypes.c_wchar_p(js_string)
+
+            result = ctypes.c_void_p()
+            err = chakraCore.JsRunScript(js_script, 0, js_source, point(result))
```
Comment (zhengrenzhe) [inline PyChakra/__init__.py:81]: it seems should be `self.chakraCore` ?

File: PyChakra/__init__.py [hunk 2]
```diff
@@ -60,31 +60,73 @@ def __init__(self):
...
 
-        # eval success
         if err == 0:
-            return True, self.__js_value_to_str(result)
+            result = self.__js_value_to_str(result)
+            if result == "undefined":
+                result = None
+            else:
+                # json => value
+                result = json.loads(result)
```
Comment (zhengrenzhe) [inline PyChakra/__init__.py:126]: missing `import json`?

**Review by zhengrenzhe — COMMENTED** [review]
thanks for your pr :D , i found two small issues.

**Comment by zhengrenzhe** [comment]
If rebase your commits can be better :D

**Comment by SeaHOH** [comment]
I am sorry for these mistakes. I am using the older version now, did not test it.

You said "rebase" is means fixed in one commit and rebase the master branch of fork?

I has another question: why do string buffer need to be larger than it was written to?
https://github.com/zhengrenzhe/PyChakra/blob/9e4d559114ebaa109430e9ecc0dd49550619d56a/PyChakra/__init__.py#L118-L130

I has try run code likes below, it works fine.
```py
result = ctypes.create_string_buffer(str_len.value) 
self.chakraCore.JsCopyString(js_value_ref, point(result), str_len.value, 0)

return result.value.decode("UTF-8")
```

**Comment by zhengrenzhe** [comment]
> You said "rebase" is means fixed in one commit and rebase the master branch of fork?

对 😃


>I has another question: why do string buffer need to be larger than it was written to?

这个写法是之前在ChakraCore的官方示例里看到的，应该是避免一些未知问题吧

**Pull request closed without merging**